cmd/logviewer/main.go

```diff
@@ -0,0 +1,261 @@
+	}
+
+	// testServer is used when a test web socket is need for testing purposes
+	testServer = cli.BoolFlag{
+		Name:  "test-server",
+		Usage: "starts a test server that will send log lines continuously. For development and debugging purposes only.",
+	}
+
+	ansiCleaner = regexp.MustCompile(ansi)
+)
+
+var log = logger.GetOrCreate("logviewer")
+var cliApp *cli.App
+var webSocket *websocket.Conn
+var manualStop chan struct{}
+var fileForLogs *os.File
+
+func main() {
+	initCliFlags()
+	manualStop = make(chan struct{})
+	log.Info("log viewer application started", "version", cliApp.Version)
+
+	cliApp.Action = func(c *cli.Context) error {
+		return startLogViewer(c)
+	}
+
+	err := cliApp.Run(os.Args)
+	if err != nil {
+		log.Error(err.Error())
+		os.Exit(1)
+	}
+}
+
+func initCliFlags() {
+	cliApp = cli.NewApp()
+	cli.AppHelpTemplate = nodeHelpTemplate
+	cliApp.Name = "Elrond Node CLI App"
+	cliApp.Version = fmt.Sprintf("%s/%s/%s-%s", "1.0.0", runtime.Version(), runtime.GOOS, runtime.GOARCH)
+	cliApp.Usage = "This is the entry point for starting a new Elrond node - the app will start after the genesis timestamp"
+	cliApp.Flags = []cli.Flag{
+		address,
+		logLevelPatterns,
+		logFile,
+		testServer,
+	}
+	cliApp.Authors = []cli.Author{
+		{
+			Name:  "The Elrond Team",
+			Email: "contact@elrond.com",
+		},
+	}
+}
+
+func startLogViewer(ctx *cli.Context) error {
+	var err error
+	logLevelPattern := ctx.GlobalString(logLevelPatterns.Name)
+	logLevels, _, err := logger.ParseLogLevelAndMatchingString(logLevelPattern)
+	if err != nil {
+		return err
+	}
+
+	if ctx.GlobalBool(testServer.Name) {
+		_ = testing.StartTestWebSocket("127.0.0.1:8080")
+		time.Sleep(time.Second * 2)
+	}
+
+	if ctx.GlobalBool(logFile.Name) {
+		err = prepareLogFile()
+		if err != nil {
+			return err
+		}
+
+		defer func() {
+			_ = fileForLogs.Close()
+		}()
+	}
+
+	addrString := ctx.GlobalString(address.Name)
+	webSocket, err = openWebSocket(addrString, logLevelPattern)
+	if err != nil {
+		return err
+	}
+	go listeningOnWebSocket()
+
+	//set this log's level to the lowest desired log level that matches received logs from elrond-go
+	lowestLogLevel := getLowestLogLevel(logLevels)
+	log.SetLevel(lowestLogLevel)
+
+	waitForUserToTerminateApp(webSocket)
+
+	return nil
+}
+
+func getLowestLogLevel(logLevels []logger.LogLevel) logger.LogLevel {
+	lowest := logLevels[0]
+	for i := 1; i < len(logLevels); i++ {
+		if lowest > logLevels[i] {
+			lowest = logLevels[i]
+		}
+	}
+
+	return lowest
+}
+
+func prepareLogFile() error {
+	var workingDir = ""
+	workingDir, err := os.Getwd()
+	if err != nil {
+		log.LogIfError(err)
+		workingDir = ""
+	}
+
+	logDirectory := filepath.Join(workingDir, defaultLogPath)
+	fileForLogs, err = core.CreateFile("logviewer", logDirectory, "log")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func openWebSocket(address string, logLevelPatterns string) (*websocket.Conn, error) {
+	u := url.URL{
+		Scheme: "ws",
```

Comment:
Is this thought of as running on the same machine as the node? Maybe there should be an option of running `wss` also

Reply:
Added option for wss